test/es-module/test-esm-non-js.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const { spawn } = require('child_process');
+const assert = require('assert');
+
+const entry = require.resolve('./test-esm-json.mjs');
```

Comment:
Double-checking: This test is expected to start failing if we unflag `.json` support? Is that dangerous since it's technically not meant to test the lack of `.json` support itself? Or, alternatively, should this test be renamed `test-esm-no-json-by-default.js`?

Reply:
The two extensions to check are `.json` and `.node`. If we unflag JSON support, then this test would need to be updated accordingly yes. I don't think that's dangerous as if/when that happens and this test fails it can be updated to handle the new scenario.

Reply:
👍 Works for me. Just wanted this documented for posterity.